07-performance.Rmd

```diff
@@ -89,6 +89,15 @@ system.time({
 identical(result1, result2)
 ```
 
+There is talk on the [R-devel email](http://r.789695.n4.nabble.com/ifelse-woes-can-we-agree-on-a-ifelse2-td4723584.html) list of speeding up `ifelse()` in base R. A simple solution is to use the `if_else()` function from **dplyr**, although, as discussed in the [same thread](http://r.789695.n4.nabble.com/ifelse-woes-can-we-agree-on-a-ifelse2-td4723584.html), it cannot replace `ifelse()` in all situations. For our exam result test example, `if_else()` works fine and is much faster than base R's implentation (although is still around 3 times slower than the hard-coded solution):
+
+```{r}
```

Comment:
I think `system.time` is perfectly acceptable for cases like this.

I tried redoing it with **microbenchmark** but it came out butt ugly:

```
microbenchmark::microbenchmark(times = 1, {
  with_ifelse = ifelse(marks >= 40, "pass", "fail"),
  
  without_ifelse = result2 = rep("fail", length(marks));
  result2[marks >= 40] = "pass"}

})
```

If you can suggest a better way of writing this I'll change it but that's the best I can come up with atm.